### PR TITLE
Fix for TLS dashboard

### PR DESCRIPTION
**Pull request by thotypous**
Setting up a TLS dashboard currently causes some issues, as initially reported in #5053. Problems were partially fixed by #5055, but some issues remain:

- Setting `dashboard_protocol` to `https` causes the dashboard to try accessing Prometheus API though https too. However, Prometheus [does not](https://prometheus.io/docs/guides/tls-encryption/) support TLS natively. Setting up a reverse proxy such as nginx would be needed for this. For now, we just fix the protocol as plain http, which is what Prometheus supports.

- People are currently using `ceph dashboard set-grafana-api-ssl-verify False` as a workaround to get Grafana to work with TLS. In order to get the certificate to validate correctly, we can allow the user to override `grafana_server_addr` with a FQDN listed in the certificate's CN or altsubjects.

Just for reference, I'm deploying with the following config:

```
dashboard_protocol: https
dashboard_admin_user: admin
dashboard_admin_password: ommited
dashboard_crt: 'ceph.crt'
dashboard_key: 'ceph.key'
grafana_admin_user: admin
grafana_admin_password: ommited
grafana_crt: 'ceph.crt'
grafana_key: 'ceph.key'
grafana_server_addr: somedomain.ufscar.br
```

### CODE REVIEW

File: group_vars/all.yml.sample [hunk 1]
```diff
@@ -768,6 +768,8 @@ dummy:
 # We only need this for SSL (https) connections
 #grafana_crt: ''
 #grafana_key: ''
+# Server hostname for which grafana_crt is valid
+#grafana_server_addr: ''
```
Comment (guits) [inline group_vars/all.yml.sample:772]: 2 things:

1/ you shouldn't edit this file directly.
please, edit `roles/ceph-defaults/defaults/main.yml` and run `./generate_group_vars_sample.sh`

2/ That will work as it is, but I wonder whether we would better setting a default value here, then we would have a condition like this on the tasks you've modified below:  

```yaml
  when:
    - grafana_server_addr != "whatever-default-value"
```

File: roles/ceph-facts/tasks/grafana.yml [hunk 2]
```diff
@@ -21,6 +23,7 @@
     grafana_server_addrs: "{{ (grafana_server_addrs | default([]) + [hostvars[item]['ansible_all_ipv4_addresses'] | ips_in_ranges(public_network.split(',')) | first]) | unique }}"
   with_items: "{{ groups.get(grafana_server_group_name, []) }}"
   when:
+    - grafana_server_addr is not defined
```
Comment (guits) [inline roles/ceph-facts/tasks/grafana.yml:26]: if you skip this task when `grafana_server_addr` is defined by the user, how `grafana_server_addrs` is supposed to be set ? It is consumed

**Comment by guits** [comment]
@fmount @fultonj @gfidente FYI

**Comment by fmount** [comment]
@thotypous how can you deal with multiple grafana instances? 
One of the main supported scenarios which is the reason to properly compute the `grafana_server_addr` fact is to allow users defining multiple grafana servers, configuring the ini file using the correct fact for the current node in which grafana is deployed.

File: roles/ceph-facts/tasks/grafana.yml (hunk 2 above)
Comment (thotypous) [reply roles/ceph-facts/tasks/grafana.yml]: Sorry, the original patch was over cae24dd85a979ed40efc7de384c158a04bd3558d and I messed it up when rebasing before sending this PR. The new patch I'm force pushing will employ a new approach.

**Comment by thotypous** [comment]
Sorry for the delay. I had several issues with the cluster I was using for tests and only got to test the patch against master a few minutes ago.

The previous approach was completely incorrect, sorry. I was originally testing against cae24dd85a979ed40efc7de384c158a04bd3558d, then took too long to send the PR and messed up when rebasing. I didn't pay attention to the new `grafana_server_addrs` fact.

@fmount Please take a look at the new approach.

**Pull request merged**